From adda69fcaaa840d9210df41fbfa788b594d78e0e Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Tue, 15 Dec 2020 20:40:04 +0800
Subject: [PATCH] ASoC: rockchip: multicodecs: Fix the build as module

ERROR: "snd_card_locked" [sound/soc/rockchip/snd-soc-rockchip-multicodecs.ko] undefined!

Fixes: 49950d606ad7 ("ASoC: rockchip: multicodecs: add support property 'rockchip,wait-card-locked'")
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
Change-Id: I71b9ec38072d1a0147ad0f9c3b6198f89aa385ec
---
 sound/soc/rockchip/rockchip_multicodecs.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/sound/soc/rockchip/rockchip_multicodecs.c b/sound/soc/rockchip/rockchip_multicodecs.c
index fb0a684d5362..ee45076718f4 100644
--- a/sound/soc/rockchip/rockchip_multicodecs.c
+++ b/sound/soc/rockchip/rockchip_multicodecs.c
@@ -148,8 +148,11 @@ static int wait_locked_card(struct device_node *np, struct device *dev)
 {
 	char *propname = "rockchip,wait-card-locked";
 	u32 cards[WAIT_CARDS];
-	int num, i;
+	int num;
 	int ret;
+#ifndef MODULE
+	int i;
+#endif
 
 	ret = of_property_count_u32_elems(np, propname);
 	if (ret < 0) {
@@ -186,6 +189,7 @@ static int wait_locked_card(struct device_node *np, struct device *dev)
 	}
 
 	ret = 0;
+#ifndef MODULE
 	for (i = 0; i < num; i++) {
 		if (!snd_card_locked(cards[i])) {
 			dev_warn(dev, "card: %d has not been locked, re-probe again\n",
@@ -194,6 +198,7 @@ static int wait_locked_card(struct device_node *np, struct device *dev)
 			break;
 		}
 	}
+#endif
 
 	return ret;
 }
-- 
2.35.3

